src/MainScreen.js

```diff
@@ -3,314 +3,375 @@ import WheelComponent from 'react-wheel-of-prizes'
-      history: [],
-      nextGameIdx: 0,
-      showPlayerSelectModal: false
-    };
-  }
-
-  changeNextGameIdx = (delta = 1) => {
-    if(this.state.nextGameIdx + delta > this.state.history.length) return false;
-    if(this.state.nextGameIdx + delta < 0) return false;
-    this.setState((state) => {
-      return {
-        ...this.state,
-        nextGameIdx: state.nextGameIdx + delta
-      }
-    });
-    return true;
-  }
-
-  moveNextGameFwd = () => {
-    return this.changeNextGameIdx();
-  }
-
-  moveNextGameBack = () => {
-    return this.changeNextGameIdx(-1);
-  }
-
-  addGameRequest = (gameObj, user) => {
-    this.setState((state) => {
-      return {
-        ...state,
-        messages: {
-          ...this.state.messages,
-          [gameObj.longName]: {
-            ...gameObj,
-            username: user,
-            time: Date.now(),
-            locked: false,
-            chosen: false
-          }
-        },
-        counter: this.state.counter + 1
-      };
-    })
-  }
-
-  toggleLock = (game) => {
-    const stateCopy = {...this.state.messages[game]};
-    stateCopy.locked = !stateCopy.locked
-
-    this.setState(() => {
-      return {
-        ...this.state,
-        messages: {
-          ...this.state.messages,
-          [game]: stateCopy
+    constructor(props){
+        super(props);
+        this.chatActivity = new ChatActivity(this.props.channel)
+        this.state = {
+            messages: {},
+            colors: randomColor({count: 99, luminosity: 'light', hue: 'blue'}),
+            counter: 0,
+            history: [],
+            nextGameIdx: 0,
+            showPlayerSelect: false
+        };
+
+        this.playerSelector = null;
+        this.messageHandler = null;
+
+        this.changeNextGameIdx = this.changeNextGameIdx.bind(this);
+        this.moveNextGameFwd = this.moveNextGameFwd.bind(this);
+        this.moveNextGameBack = this.moveNextGameBack.bind(this);
+        this.addGameRequest = this.addGameRequest.bind(this);
+        this.toggleLock = this.toggleLock.bind(this);
+        this.setNextGame = this.setNextGame.bind(this);
+        this.addGameToQueue = this.addGameToQueue.bind(this);
+        this.onWheelSpun = this.onWheelSpun.bind(this);
+        this.removeGame = this.removeGame.bind(this);
+        this.onMessage = this.onMessage.bind(this);
+        this.togglePlayerSelect = this.togglePlayerSelect.bind(this);
+        this.routePlayRequest = this.routePlayRequest.bind(this);
+        this.routeLeaveRequest = this.routeLeaveRequest.bind(this);
+        this.routeOpenQueueRequest = this.routeOpenQueueRequest.bind(this);
+        this.routeCloseQueueRequest = this.routeCloseQueueRequest.bind(this);
+        this.routeClearQueueRequest = this.routeClearQueueRequest.bind(this);
+        this.startGame = this.startGame.bind(this);
+        this.setMessageHandlerRef = this.setMessageHandlerRef.bind(this);
+        this.setPlayerSelectRef = this.setPlayerSelectRef.bind(this);
```

Comment:
wait, why all the explicit binds when we're also using lambdas?

Reply:
Eh, I figured it'd help with any state and/or scope issues. We use 'em at work, and it made sense for me at the time, lol.